unison-cli/src/Unison/Codebase/Editor/HandleCommand.hs

```diff
@@ -118,8 +134,11 @@ commandLine config awaitInput setBranchRef rt notifyUser notifyNumbered loadSour
     Typecheck ambient names sourceName source -> do
       -- todo: if guids are being shown to users,
       -- not ideal to generate new guid every time
-      i <- State.get
-      State.modify' (+1)
+      iVar <- ask
```

Comment:
TBH I'm not exactly sure why we're going to so much work to get a seed for randomness; seems like there might be better ways to do this.

Reply:
The runs are deterministic to produce stable transcript output — seed changes repeatably for each typecheck. What are some better ways to do this?

Reply:
Ah that makes sense. I'm not too worried about this tbh, it's a pretty small thing 👍🏼